pkg/api/v1alpha1/edgeconnect/edgeconnect_types.go

```diff
@@ -166,6 +166,7 @@ func (dk *EdgeConnectStatus) SetPhase(phase status.DeploymentPhase) bool {
 // +kubebuilder:printcolumn:name="ApiServer",type=string,JSONPath=`.spec.apiServer`
 // +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.phase`
 // +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:validation:XValidation:rule="size(self.metadata.name) < 40",reason="FieldValueInvalid",fieldPath=".metadata",message="The length limit for the name of a EdgeConnect is 40, because it is the base for the name of resources related to the EdgeConnect. The limit is necessary because kubernetes uses the name of some resources for the label value, which has a limit of 63 characters."
```

Comment:
it's not accurate, previously it was max was 40, so `40 > 40` is false, it means it will show error only on 41, but now we show error on 40. May be it's want we want, but it's new behaviour.

Reply:
so you should change it to `<=` or `< 41`